src/fluent.rs

```diff
@@ -160,6 +196,36 @@ impl Localizer {
     }
 }
 
+pub trait MessageKey {
+    fn key(&self) -> &str;
+
+    fn attribute(&self) -> Option<&str> {
+        None
+    }
+}
+
+impl MessageKey for str {
+    fn key(&self) -> &str {
+        self
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+pub struct MessageAttribute<'key, 'attribute> {
+    pub key: &'key str,
+    pub attribute: &'attribute str,
+}
+
+impl<'key, 'attribute> MessageKey for MessageAttribute<'key, 'attribute> {
+    fn key(&self) -> &str {
+        self.key
+    }
+
+    fn attribute(&self) -> Option<&str> {
+        Some(self.attribute)
+    }
+}
+
 #[cfg(test)]
 mod tests {
     use super::*;
```

Comment:
@jreppnow 
How about we add some simple tests to show that you can provide both normal keys and attribute keys as params in the Localizer::format_message function? Could help ensure we don't break this capability and prove that we have backwards compatibility.

Reply:
Tests added! Seems to work well!